fix UnicodeDecodeError

glowinthedark opened the pull request:
fix UnicodeDecodeError for MDX dictionaries that use utf-8 instead of utf-16 for metadata

Commit:
fix UnicodeDecodeError

fix UnicodeDecodeError for MDX dictionaries that use utf-8 instead of utf-16 for metadata

liuyug commented:
```json.detect_encoding ```  ? Is it a function? 
Which dictionary used utf-8 encoding? Which APP compiled the dictionary?

glowinthedark committed:
skip invalid dictionaries

allow to continue and prevent app failure even if indexing fails for a single file

liuyug commented:
1. ```json.detect_encoding``` is non-public function. 

2. self._encoding was taken in MDX header and it is ```UTF-16``` in MDD file.
check ```Encoding```  item in header of MDX file. The default value is 'utf-8'.
